File: javascript/6_kyu/persistent_bugger.js
// Persistent Bugger
// https://www.codewars.com/kata/55bf01e5a717a0d57e0000ec/train/javascript

function multiply(n) {
    var n_list = n.toString().split("").map(Number);

    return n_list.reduce( function(a, b) { return a * b } );
}

function persistence(num) {
    var count = 0;

    while ( num >= 10 ) {
        count++;
        var num = multiply(num);
    }

    return count;
}

console.log(persistence(999))

File: python/6_kyu/persistent_bugger.py
# Persistent Bugger
# https://www.codewars.com/kata/55bf01e5a717a0d57e0000ec/train/python

def multiply(num):
    num_list = [int(x) for x in list(str(num))]
    result = 1

    for i in num_list:
        result = result * i
    
    return result


def persistence(n):
    count = 0

    while len(str(n)) >= 2:
        count += 1
        n = multiply(n)

    return count

print(persistence(999))